### PR TITLE
Add admissions from supervision to prison by district chart

### LINKED ISSUE #1451: [US_TN] Admissions from supervision to prison by district chart
**Is your feature request related to a problem? Please describe.**
Add a "Admissions from supervision to prison by district" chart

metric should be called `supervisionToPrisonPopulationByDistrict`, and will be an instance of `PopulationSnapshotMetric`.
      id: "supervisionToPrisonPopulationByDistrict",
      tenantId: this.rootStore.currentTenantId,
      compartment: "SUPERVISION",
      sourceFilename: "supervision_to_prison_population_snapshot_by_dimension",
      rootStore: this.rootStore,
      dataTransformer: createDistrictPopulationSnapshot,
      enabledFilters: [FILTER_TYPES.GENDER, FILTER_TYPES.DISTRICT],
      enabledMoreFilters: [FILTER_TYPES.AGE_GROUP, FILTER_TYPES.LEGAL_STATUS],

You can use `VizPopulationSnapshot`, and modify the `PopulationSnapshotMetric` to have an `accessor` attribute which specifies which attribute/dimension is displayed in the chart. In this case, `accessor: district`.

[Design](https://www.figma.com/file/Mw6kSEP2jdr9i7UVEjPuCI/UP-Consolidation)

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by nasaownsky**
## Description of the change

Adds admissions from supervision to prison by district chart

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes #1451 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by nasaownsky** [comment]
hey, @jovergaag! I've been struggling for a while now, and I'm asking for help or advice. I try to make `PopulationSnapshotMetric` metric more general and suitable for different accessors, and even had some progress with that, but had fall into problem where data from `supervision_to_prison_population_snapshot_by_dimension` does not fetch at all. So I've made separate `PopulationSnapshotSupervisionMetric` trying to debug this issue. And it seems to be completely legit implementation, but the data isn't displaying! Argh! Maybe I've missed something? Could you please look at with your fresh eyes?

File: src/core/models/utils.ts [hunk 1]
```diff
@@ -48,14 +49,31 @@ export function createProjectionTimeSeries(
...
       ageGroup: record.age_group as AgeGroup,
       facility: record.facility,
-      totalPopulation: parseInt(record.count),
+      count: parseInt(record.count),
+      lastUpdated: formatDateString(record.last_updated),
+    };
+  });
+}
+
+export function createDistrictPopulationSnapshot(
```
Comment (jovergaag) [inline src/core/models/utils.ts:65]: Let's rename this to `createSupervisionPopulationSnapshot` and rename line 50 to `createPrisonPopulationSnapshot`

File: src/core/VizPopulationSnapshot/VizPopulationSnapshot.tsx [hunk 2]
```diff
@@ -99,22 +105,22 @@ const VizPopulationSnapshot: React.FC<VizPopulationOverTimeProps> = ({
...
             }
             return { fill: styles.dataForest };
           }}
           rAccessor="value"
           rExtent={yRange}
           // @ts-ignore
-          oLabel={(facility: string, _: any) => {
-            return <text textAnchor="middle">{facility}</text>;
+          oLabel={(accessorValue: string, _: any) => {
+            return <text textAnchor="middle">{accessorValue}</text>;
```
Comment (jovergaag) [inline src/core/VizPopulationSnapshot/VizPopulationSnapshot.tsx:123]: The district x-axis  labels are showing up as `DISTRICT_1` etc. Please format them such that they match the label in the filter options.

**Review by jovergaag — CHANGES_REQUESTED** [review]
Sorry about that bad fixture file, I hope you didn't spend too much time on it! I sent a new file, give it a try.

I also pushed up a commit to get the chart populating and one for the synced content.

File: src/core/models/SupervisionPopulationSnapshotMetric.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+import map from "lodash/fp/map";
+import sumBy from "lodash/fp/sumBy";
+import values from "lodash/fp/values";
+
+import { downloadChartAsData } from "../../utils/downloads/downloadData";
+import { DownloadableData, DownloadableDataset } from "../PagePractices/types";
+import PathwaysMetric, { BaseMetricConstructorOptions } from "./PathwaysMetric";
+import { SupervisionPopulationSnapshotRecord } from "./types";
+
+export default class SupervisionPopulationSnapshotMetric extends PathwaysMetric<SupervisionPopulationSnapshotRecord> {
```
[the remark this answers is not in the document]
Comment (nasaownsky) [reply src/core/models/SupervisionPopulationSnapshotMetric.ts:31]: Yeah, I don't love neither. If I can come up with some elegant solution that could simplify things, I'll push it within next PR.

File: src/core/models/SupervisionPopulationSnapshotMetric.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    const datasets = [] as DownloadableDataset[];
+    const data: Record<string, number>[] = [];
+    const labels: string[] = [];
+
+    this.dataSeries.forEach((d: SupervisionPopulationSnapshotRecord) => {
+      data.push({
+        Count: Math.round(d.count),
+      });
+
+      labels.push(d[this.accessor].toString());
```
Comment (jovergaag) [inline src/core/models/SupervisionPopulationSnapshotMetric.ts:97]: We also need to convert this to the filter option label. 

Screen Shot 2021-11-30 at 10 48 44 AM

File: src/core/models/PrisonPopulationSnapshotMetric.ts [hunk 5]
```diff
@@ -67,18 +64,18 @@ export default class PopulationSnapshotMetric extends PathwaysMetric<PopulationS
...
+      groupBy((d: PrisonPopulationSnapshotRecord) => [d[this.accessor]]),
       values,
       map((dataset) => ({
         gender: dataset[0].gender,
         legalStatus: dataset[0].legalStatus,
         facility: dataset[0].facility,
         lastUpdated: dataset[0].lastUpdated,
         ageGroup: dataset[0].ageGroup,
-        totalPopulation: sumBy("totalPopulation", dataset),
+        count: sumBy("count", dataset),
```
Comment (jovergaag) [inline src/core/models/PrisonPopulationSnapshotMetric.ts:75]: Thank you for cleaning this up to be more consistent!

**Review by jovergaag — CHANGES_REQUESTED** [review]
One small change then we're all set! Once fixed I will merge this into the base branch and we will land it all together when the schema change is in.

**Review by jovergaag — APPROVED** [review]

**Pull request merged**